### PR TITLE
Update Bitmap::len to return bits rather than bytes

### LINKED ISSUE #730: `Bitmap::len` returns the number of bytes, not bits.
This is confusing and leads to all kinds of bugs to downstream users.

### PR DESCRIPTION
**Pull request by matthewmturner**
# Which issue does this PR close?

Closes #730 

# Rationale for this change
 
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by jorgecarleitao** [comment]
Looks a good start. I think that every `bitmap.len()` out there needs to be divided by 8 now, since this is a backward-incompatible change.

**Review by jorgecarleitao — APPROVED** [review]

**Comment by matthewmturner** [comment]
The errors were actually just because i needed to add the submodules.

I see you already approved - do you still need me to check if `bitmap.len` is used anywhere?

**Comment by jorgecarleitao** [comment]
No, checks pass, it should be good to go.

File: arrow/src/bitmap.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@ impl Bitmap {
     }
 
     pub fn len(&self) -> usize {
```
Comment (alamb) [inline arrow/src/bitmap.rs:47]: Can we also add a docstring here that explains what `Bitmap::len` actually returns (aka the length in bits, not bytes)?

**Review by alamb — COMMENTED** [review]
Thank you @matthewmturner  -- I think this does what is called for in https://github.com/apache/arrow-rs/issues/744 👍 

@jorgecarleitao  perhaps you can have a look as well?

I also looked at the uses of `Bitmap::len` in the arrow crate and it only appears to be used in tests:

```
/Users/alamb/Software/arrow-rs/arrow/src/bitmap.rs
120:         assert_eq!(64, Bitmap::new(63 * 8).len());
121:         assert_eq!(64, Bitmap::new(64 * 8).len());
122:         assert_eq!(128, Bitmap::new(65 * 8).len());
```

One concern I have is related to backwards compatibility -- if we change `len` to return the length in bits but change nothing else, the `rustc` compiler will not alert any users of this function that the API has changed.

What about also renaming `Bitmap::len()` to `Bitmap::byte_len()` to make it super clear what is going on as well as generating compiler errors on upgrade?

**Comment by jorgecarleitao** [comment]
I would not consider this backward incompatibility but a bug fix: imo `Bitmap::len` should be in bits, not bytes. We actually found this while assuming that the len was in bits (it is a bitmap, after all).

**Review by alamb — APPROVED** [review]

**Pull request merged**